### PR TITLE
fix(standalone): hide etcd password in logs

### PR DESCRIPTION
**Pull request by kwannoel**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Fixes https://github.com/risingwavelabs/risingwave/issues/13005.

~A quick hack to hide the pwd, without having to introduce a full custom display trait.~

Hides the etcd password by wrapping the field in `redact::Secret`.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Review by Li0k — APPROVED** [review]

File: src/cmd_all/src/standalone.rs
Comment (xxchan) [inline src/cmd_all/src/standalone.rs:1]: IIRC this will also be printed for non standalone mode?

Comment (xxchan) [reply src/cmd_all/src/standalone.rs:1]: I think this hack (or solution?) can be safer
```rust
#[derive(Educe)]
#[educe(Debug)]
pub struct MetaNodeOpts {
	// ...

    #[clap(long, env = "RW_ETCD_PASSWORD", default_value = "")]
	#[educe(Debug(method = "redact_secret"))]
    etcd_password: String,
}

fn redact_secret(_s: &u8, f: &mut Formatter) -> fmt::Result {
    f.write_str("******")
}
```

Comment (BugenZhao) [reply src/cmd_all/src/standalone.rs:1]: Well, I believe there's a type-safer way to achieve this.

https://crates.io/crates/redact

File: src/meta/node/src/lib.rs [hunk 1]
```diff
@@ -196,7 +198,10 @@ pub fn start(opts: MetaNodeOpts) -> Pin<Box<dyn Future<Output = ()> + Send>> {
                     .map(|x| x.to_string())
                     .collect(),
                 credentials: match opts.etcd_auth {
-                    true => Some((opts.etcd_username, opts.etcd_password)),
+                    true => Some((
+                        opts.etcd_username,
+                        opts.etcd_password.expose_secret().to_string(),
```
Comment (xxchan) [inline src/meta/node/src/lib.rs:203]: Hmmm, I feel ***ideally*** we should make all the intermediate stuff `Secret`, and reveal secret until we have to. Just checked `MetaStoreBackend` also `#[derive(Debug)]`. 🤣

Just raise this point. Not a hard requirement for now.

**Review by xxchan — APPROVED** [review]

File: src/meta/node/src/lib.rs (hunk 1 above)
Comment (kwannoel) [reply src/meta/node/src/lib.rs:203]: Good point.

Comment (kwannoel) [reply src/meta/node/src/lib.rs:203]: https://github.com/risingwavelabs/risingwave/issues/12673#issuecomment-1780729326

**Review by BugenZhao — APPROVED** [review]

**Pull request merged**